Calibration/HcalAlCaRecoProducers/plugins/AlCaHcalIsotrkFilter.cc

```diff
@@ -0,0 +1,183 @@
+// system include files
+#include <atomic>
+#include <memory>
+#include <cmath>
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDFilter.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/Run.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+
+#include "DataFormats/Common/interface/Handle.h"
+#include "DataFormats/Common/interface/Ref.h"
+#include "DataFormats/HcalCalibObjects/interface/HcalIsoTrkCalibVariables.h"
+
+//#define EDM_ML_DEBUG
+//
+// class declaration
+//
+
+namespace alcaHcalIsoTrk {
+  struct Counters {
+    Counters() : nAll_(0), nGood_(0) {}
+    mutable std::atomic<unsigned int> nAll_, nGood_;
+  };
+}  // namespace alcaHcalIsoTrk
+
+class AlCaHcalIsotrkFilter : public edm::stream::EDFilter<edm::GlobalCache<alcaHcalIsoTrk::Counters> > {
```

Comment:
By quick look this EDFilter looks like it would be simpler to implement as `edm::global::EDFilter` and all the counters as `mutable std::atomic<unsigned in>`.

Reply:
Done